src/components/modules/toolbar/inline.ts

```diff
@@ -393,11 +414,16 @@ export default class InlineToolbar extends Module {
 
     /**
      * Filter buttons if Block Tool pass config like inlineToolbar=['link']
+     * Else filter them according to the default inlineTOllbar property.
      */
     if (Array.isArray(inlineToolbarSettings)) {
       buttons.forEach((button) => {
         button.hidden = !inlineToolbarSettings.includes(button.dataset.tool);
       });
+    } else if (Array.isArray(this.config.inlineToolbar)) {
+      buttons.forEach((button) => {
+        button.hidden = !this.config.inlineToolbar.includes(button.dataset.tool);
+      });
     }
```

Comment:
I'm not sure, but the lines above could be simplified. You're repeating yourself: the difference are condition and iterable object. Maybe create a new private function and pass buttons and hiddenList

Reply:
Yes, to be honest, your thoughts did cross my mind but I couldn't think of a way to implement it. I will try to implement it as you have suggested.

Reply:
Can you check the recent commit and suggest more optimizations?